built-in-services/account/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+use crate::types::{
+    Account, GenerateAccountPayload, GenerateAccountResponse, GetAccountPayload, PayloadAccount,
+    Permission, VerifyPayload, VerifyResponse, Witness, ACCOUNT_TYPE_MULTI_SIG,
+    ACCOUNT_TYPE_PUBLIC_KEY, MAX_PERMISSION_ACCOUNTS,
+};
+use binding_macro::{cycles, service};
+use bytes::Bytes;
+use common_crypto::{Crypto, Secp256k1};
+use hasher::{Hasher, HasherKeccak};
+use protocol::traits::{ExecutorParams, ServiceResponse, ServiceSDK};
+use protocol::types::{Address, Hash, Hex, ServiceContext};
+
+#[cfg(test)]
+mod tests;
+pub mod types;
+
+pub struct AccountService<SDK> {
+    sdk: SDK,
+}
+
+#[service]
+impl<SDK: ServiceSDK> AccountService<SDK> {
+    pub fn new(sdk: SDK) -> Self {
+        Self { sdk }
+    }
+
+    #[cycles(100_00)]
+    #[read]
+    fn verify_signature(
+        &self,
+        ctx: ServiceContext,
+        payload: VerifyPayload,
+    ) -> ServiceResponse<VerifyResponse> {
+        let wit_str = String::from_utf8(payload.witness.to_vec());
+        if wit_str.is_err() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                112,
+                "witness from utf8 bytes to string failed".to_owned(),
+            );
+        }
+
+        let wit_res: Result<Witness, _> = serde_json::from_str(wit_str.unwrap().as_str());
+        if wit_res.is_err() {
+            return ServiceResponse::<VerifyResponse>::from_error(
+                112,
+                "witness not valid".to_owned(),
+            );
+        }
+
+        let wit = wit_res.unwrap();
+        if wit.signature_type == ACCOUNT_TYPE_PUBLIC_KEY {
```

Comment:
if you don't use multiSig ever, you'll be confused about SINGLE_SIGN.
ACCOUNT_TYPE_PUBLIC_KEY means the account is generated from public_key, it may be natural for blockchain user?